app/controllers/api/base_controller.rb

```diff
@@ -1,16 +1,44 @@
 # Base controller for OFN's API
-# Includes the minimum machinery required by ActiveModelSerializers
+require_dependency 'spree/api/controller_setup'
+
 module Api
-  class BaseController < Spree::Api::BaseController
-    # Need to include these because Spree::Api::BaseContoller inherits
-    # from ActionController::Metal rather than ActionController::Base
-    # and they are required by ActiveModelSerializers
+  class BaseController < ActionController::Metal
+    include Spree::Api::ControllerSetup
+    include Spree::Core::ControllerHelpers::SSL
+    include ::ActionController::Head
+
+    respond_to :json
+
+    attr_accessor :current_api_user
+
+    before_filter :set_content_type
+    before_filter :authenticate_user
+    after_filter  :set_jsonp_format
+
+    rescue_from Exception, with: :error_during_processing
+    rescue_from CanCan::AccessDenied, with: :unauthorized
+    rescue_from ActiveRecord::RecordNotFound, with: :not_found
+
+    helper Spree::Api::ApiHelpers
+
+    ssl_allowed
+
+    # Include these because we inherit from ActionController::Metal
+    #   rather than ActionController::Base and these are required for AMS
```

Comment:
I'd be great to understand why and document it, maybe something for a new spike issue.

Reply:
these lines of code were already there and this new comment is basically an improvement from the removed top class comment "# Includes the minimum machinery required by ActiveModelSerializers"

I dont know exactly why they are needed but it looks like they are, as mentioned, "the minimum machinery required by ActiveModelSerializers"

Do you think this is not needed?

Reply:
yes, Metal is a stripped down version for apis, 40% performance improvement in some cases according to this https://stackoverflow.com/questions/18024337/what-actually-does-rails-actioncontrollermetal